app/environments/brassbirmingham/brassbirmingham/envs/classes/board.py

```diff
@@ -157,26 +178,58 @@ def areNetworked(
     def removeXCoal(self, X: int, towns: List[Town], player: Player):
         for town in towns:
             availableCoal = self.getAvailableCoalBuildingsTradePosts(town)
-            if availableCoal == 0:
+            if len(availableCoal) == 0:
                 continue
 
             _available = availableCoal.pop(
                 0
-            )  # todo assert these changes to resourceAmount are actually affecting the building
-            # take coal away from resources
+            )
             while X > 0:
                 if _available.type == "TradePost":
                     cost = self.priceForCoal(X)
-                    player.money -= cost  # todo assert this changes player's money
+                    player.pay(cost)
                     self.coalMarketRemaining -= X
                     return
                 else:
                     _available.decreaseResourceAmount(1)
                     if _available.resourceAmount == 0:
+                        if len(availableCoal) == 0:
+                            assert X == 1
+                            return
                         _available = availableCoal.pop(0)
                 X -= 1
             return
 
+    """
+    removeXIron
+    
+    :param X: amount of iron to remove
+    :param player: player to remove money from (if necessary)"""
+
+    def removeXIron(self, X: int, player: Player):
+        availableIron = self.getAvailableIronBuildingsTradePosts()
+        if len(availableIron) == 0:
```

Comment:
I guess we haven't been exactly clear on how we handle when the game tries to do something illegal. Should we assert False here? Or maybe return False?

Reply:
I would raise some kind of error yeah